NuRadioMC/utilities/inelasticities.py

```diff
@@ -23,58 +37,112 @@ def get_neutrino_inelasticity(n_events, rnd=None):
     ----------
     n_events: int
         Number of events to be returned
+    model: string
+        the inelasticity model to use
     rnd: random generator object
         if None is provided, a new default random generator object is initialized
     Returns
     -------
     inelasticies: array
         Array with the inelasticities
     """
-    if(rnd is None):
-        rnd = np.random.default_rng()
-    R1 = 0.36787944
-    R2 = 0.63212056
-    inelasticities = (-np.log(R1 + rnd.uniform(0., 1., n_events) * R2)) ** 2.5
+    rnd = rnd or np.random.default_rng()
+
+    if model.lower() == "ctw":
+        # based on shelfmc
+        R1 = 0.36787944
+        R2 = 0.63212056
+        return (-np.log(R1 + rnd.uniform(0., 1., n_events) * R2)) ** 2.5
+
+    elif model.lower() == "bgr18" or model.lower() == "hedis_bgr18":
+        yy = np.zeros(n_events)
+        uEE = np.unique(nu_energies)
+        uFlavor = np.unique(flavors)
+        uNCCC = np.unique(ncccs)
+        for energy in uEE:
+            if energy > 10 * units.EeV:
+                logger.warning("You are requesting inelasticities for energies outside of the validity of the BGR18 model. "
+                               f"You requested {energy / units.eV:.2g}eV. Largest available energy is 10EeV, returning result for 10EeV.")
+
+            for flavor in uFlavor:
+                for nccc in uNCCC:
+                    mask = (nu_energies == energy) & (flavor == flavors) & (nccc == ncccs)
+                    size = n_events
+                    if isinstance(mask, np.ndarray):
+                        size = np.sum(mask)
+
+                    nccc = nccc.upper()
+                    iF = np.argwhere(flavors_ref == flavor)[0][0]
+                    inccc = np.argwhere(ncccs_ref == nccc)[0][0]
+                    iE = np.argmin(np.abs(energy - nu_energies_ref))
+                    get_y = intp.interp1d(np.log10(yy_ref), np.log10(dsigma_dy_ref[iF, inccc, iE]),
+                                          fill_value="extrapolate", kind="cubic")
+
+                    yyy = np.linspace(0, 1, 1000)
```

Comment:
for BGR I initially thought this quantisation here might lead to the binning effect... but for CTW this is not used...